conans/client/conf/__init__.py

```diff
@@ -22,12 +22,14 @@
         version: ["7.0", "7.1", "8.0", "8.1", "8.2", "8.3", "9.0", "9.1", "9.2", "9.3", "10.0", "10.1", "10.2", "10.3"]
     FreeBSD:
     SunOS:
-arch: [x86, x86_64, ppc64le, ppc64, armv6, armv7, armv7hf, armv8, sparc, sparcv9, mips, mips64]
+    Arduino:
+        board: ANY
```

Comment:
Hi @memsharded and @lasote, thanks for your feedback.

## 1. Regarding the `board` field.
This field is used to select correct board by the toolchain and is case sensitive, so Uno and uno are 2 different boards and if you do NOT have board named Uno registered, build will fail.  So typos will be uncovered during the build. I am referring to CMake toolchain https://github.com/Dr-QP/conan-arduino-toolchain.

New [arduino-builder toolchain](https://github.com/arduino/arduino-builder) has argument 
> -fqbn: Mandatory. Fully Qualified Board Name, e.g.: arduino:avr:uno` 

which matches definitions in conan: OS:ARCH:BOARD

Boards are defined in the files like this one https://github.com/Dr-QP/arduino-board-xyzrobot/blob/develop/XYZrobot/boards.txt (This repo defines board hardware as conan package). 

## 2. Regarding `Arduino` as OS.
It should be OS as majority Arduino projects are for Arduino only and depend on its system headers and libraries and auto-magic library linking and method forward declarations.

> An Arduino sketch differs from a standard C program in that it misses a main (provided by the Arduino core), function prototypes are not mandatory, and libraries inclusion is automagic (you just have to #include them). This tool generates function prototypes and gathers library paths, providing gcc with all the needed -I params.

If developer decides to have OS agnostic package he can always override `package_id` and have something like this:
```
    def package_id(self):
        self.info.settings.os = ""
        self.info.settings.os.board = ""
        self.info.settings.arch = ""
        self.info.settings.compiler = ""
        self.info.settings.compiler.version = ""
```